llvm/utils/git/code-format-helper.py

```diff
@@ -115,24 +115,12 @@ class ClangFormatHelper(FormatHelper):
     def instructions(self) -> str:
         return " ".join(self.cf_cmd)
 
-    @cached_property
-    def libcxx_excluded_files(self) -> list[str]:
-        with open("libcxx/utils/data/ignore_format.txt", "r") as ifd:
-            return [excl.strip() for excl in ifd.readlines()]
-
-    def should_be_excluded(self, path: str) -> bool:
-        if path in self.libcxx_excluded_files:
-            print(f"{self.name}: Excluding file {path}")
-            return True
-        return False
-
     def filter_changed_files(self, changed_files: list[str]) -> list[str]:
         filtered_files = []
         for path in changed_files:
             _, ext = os.path.splitext(path)
             if ext in (".cpp", ".c", ".h", ".hpp", ".hxx", ".cxx"):
-                if not self.should_be_excluded(path):
-                    filtered_files.append(path)
+                filtered_files.append(path)
```

Comment:
Does this work for headers without any extension?

Reply:
@tru Did we ever implement the changes so that the `clang-format` job would run on extension-less headers in libc++? If not, this means that we actually *haven't* been running the `clang-format` checking on headers like `__config`, which would explain a whole lot of the confusion we've been having.

Reading through the code, I think the answer is "no, it slipped through the cracks and was never implemented".

Reply:
Assuming the answer is "no", I suggest we start by shipping https://github.com/llvm/llvm-project/pull/73142.

Reply:
We probably should also include '.inc' and '.cppm' files. These are both ued form modules and should be formatted at the moment.

Reply:
Let's take this to #73142.